examples/csv_to_schemaorg.py

```diff
@@ -1,37 +1,55 @@
+#!/usr/bin/env python3
+
 import os
+import re
 import json
+import argparse
 
 import pandas as pd
 
 from schematic.schemas.explorer import SchemaExplorer
-
 from schematic.utils.csv_utils import create_schema_classes
 from schematic.utils.config_utils import load_yaml
 
-from definitions import CONFIG_PATH, DATA_PATH
-
-config_data = load_yaml(CONFIG_PATH)
-
-# path to base schema
-base_schema_path = os.path.join(DATA_PATH, '', config_data["model"]["biothings"]["location"])
-
-# schema name (used to name schema json-ld file as well)
-output_schema_name = "HTAN"
-
-# schema extension definition csv files
-schema_extensions_csv = [
-                        os.path.join(DATA_PATH, '', 'csv/HTAN.csv')
-                        ]
+# Constants (to avoid magic numbers)
+FIRST = 0
+
+# Try to find BioThings JSON-LD file relative to script
+script_dir = os.path.dirname(os.path.realpath(__file__))
+biothings  = os.path.join(script_dir, "..", "data", "schema_org_schemas",
```

Comment:
Great PR @BrunoGrandePhD , thank you!

Quick question here, any particular reason why you've chosen to go with relative paths? Our philosophy behind creating and using the `config_data` object was that it would load all the configurations from the `config.yml` file and would save us the trouble of modifying any hardcoded paths within any of the scripts.

My thoughts on the above lines are that if ever we decide to change the names of the folders within the `data/` then we would have to change it in this `csv_to_schemaorg.py` file as well, in addition to changing it in the `config.yml` file.

Reply:
Moving away from using `config_data` was the change I was most hesitant about committing. The reason why I opted to go ahead with the change was the fact that the script imports from the `definitions.py` module, which assumes that the script is being run in the root of the `schematic` repo. Prompted by your question here, I just tried running the script in a different directory and surprisingly, it worked. 

Having looked into this further, the only reason why this has worked so far is because using `pip install -e` adds the repo directory to the Python module path (_i.e._ `import sys; print(sys.path)`). As a result, the `definitions` module can be imported. If you install `schematic` without the `-e` option, running the script results in a `ModuleNotFoundError: No module named 'definitions'` error. 

Maybe we can still rely on the config file but use a path provided at the command line instead of relying on the `definitions` module. What do you think, @sujaypatil96?

Reply:
Right, so `definitions.py` was created with the intent that it would be the file that included all the "constants". All constants that are currently hardcoded (in the `examples/` folder) should be ideally live in that file. `definitions.py` was inspired by Django because that's how Django maintains constants.

> Having looked into this further, the only reason why this has worked so far is because using `pip install -e .` adds the repo directory to the Python module path (i.e. `import sys; print(sys.path)`). As a result, the `definitions` module can be imported. If you install `schematic` without the -e option, running the script results in a `ModuleNotFoundError: No module named 'definitions' error`.

Yup, you're exactly right, and makes sense. But this `definitions.py` file contains constants that are used by most of the example files in the project, do you think it would be a good idea to simply move the `definitions.py` file into the `schematic` package as well and maintain it there? What do you think about that approach @BrunoGrandePhD?

Reply:
I've just got off a call with @sujaypatil96. Here's a quick summary of our conclusions:

- Given that the goal is to submit the schematic package to PyPI, the current implementation of the `definitions.py` script will be problematic because it will refer to file paths within the Python installation (where packages are installed). This assumes that `definitions.py` gets moved within the schematic package. This is because of the use of the `__file__ special attribute in `definitions.py`. 
- It's conceivable that a user might have one installation of schematic (installed from PyPI) but multiple projects to manage. To handle this case, all file paths that are project specific (such as the various account credentials) need to be user-configurable. Currently, they are relative to the package, and thus only only project could be managed per schematic installation. 
- To address these issues, the default values in `definitions.py` will be moved to a section in the `config.yml` file. This section will be annotated with a comment along the lines of "Don't edit this section unless you know what you're doing". Additionally, every user-facing script in `examples/` (which will eventually be exposed as command-line tools when installed via pip) will have a `--config` argument to specify the project configuration YAML file. 

I'm going to take a first stab at this and submit a PR to solicit additional comments on the design.